maestro-client/src/main/java/maestro/ios/GetRunningAppIdResolver.kt

```diff
@@ -0,0 +1,60 @@
+package maestro.ios
+
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import maestro.debuglog.DebugLogStore
+import okhttp3.HttpUrl
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import java.util.concurrent.TimeUnit
+
+class GetRunningAppIdResolver {
+    private val okHttpClient by lazy {
+        OkHttpClient.Builder()
+            .connectTimeout(10, TimeUnit.SECONDS)
+            .readTimeout(30, TimeUnit.SECONDS)
+            .build()
+    }
+
+    private val logger = DebugLogStore.loggerFor(GetRunningAppIdResolver::class.java)
+
+    fun getRunningAppId(): String {
+        val gson = Gson()
+        val appIds = GetRunningAppRequest(Simctl.listApps())
+
+        logger.info("installed apps: $appIds")
+
+        val mediaType = "application/json; charset=utf-8".toMediaType()
+        val body = gson.toJson(appIds).toRequestBody(mediaType)
+
+        val httpUrl = HttpUrl.Builder()
+            .scheme("http")
+            .host("localhost")
+            .addPathSegment("runningApp")
+            .port(9080)
+            .build()
+        val request = Request.Builder()
+            .addHeader("contentType", "application/json")
+            .url(httpUrl)
+            .post(body)
+            .build()
+        val response = okHttpClient.newCall(request).execute()
```

Comment:
Lets close the response by `use` to close the response body. See the view hierarchy call for implementation.